gitensite/apps/bookinfo/db.py

```diff
@@ -11,6 +11,9 @@ def addBookFromYaml(yaml):
 
     (book,created) = Book.objects.get_or_create(book_id=int(obj['identifiers']['gutenberg']))
 
+    if "repo_name" in obj:
```

Comment:
It's "_repo", not "repo_name". See https://github.com/gitenberg-dev/documentation/blob/master/metadata/pandata_attribute_dictionary.yaml

Reply:
The repo_name property comes the load_repos management command, where I add it to the end of the YAML data. See the changes to load_repos.py in this PR. This is how I pass the repo name to the addBookFromYaml function. There doesn't appear to be a "_repo" property that exists on the object created by the BookMetadata constructor. Let me know if I am misunderstanding what you are saying.

Reply:
gitenberg.util.catalog.BookMetadata objects are meant to be constructed with gitenberg.book.Book objects. The constructor for these objects takes repo_name as a parameter, if not, it generates the reponame. Your load_repos is constructing with it gitensite.apps.bookinfo.models.Book objects. If you want to duck type it, you need your Book object to mimic the behaviour of the gitenberg.book.Book object.